### PR TITLE
Use built-in AWS credentials loading system

### PR DESCRIPTION
**Pull request by dobesv**
This just lets `aws-sdk` load the credentials in the default way, this is more flexible than the non-standard env vars being used before.

### CODE REVIEW

**Comment by fox1t** [comment]
This is a major version bump. @tehKapa Let's prepare a new release and then we can work on the npx thing.

**Comment by fox1t** [comment]
@dobesv thanks for proposing and fixing this.

File: src/plugins/remote-cache/storage/s3.ts [hunk 1]
```diff
@@ -2,25 +2,22 @@ import s3 from 's3-blob-store'
...
   bucket: string
 }
-export function createS3({ accessKey, secretKey, bucket, region, endpoint }: S3Options) {
-  if (!accessKey || !secretKey || !(region || endpoint)) {
-    throw new Error(
-      `To use S3 storage "accessKey (S3_ACCESS_KEY)", "secretKey (S3_SECRET_KEY)", and one of "region (S3_REGION)" and "endpoint (S3_ENDPOINT)" parameters are mandatory.`,
-    )
+export function createS3({ bucket, endpoint }: S3Options) {
+  // Propagate "old" environment variables names to new ones for backwards compatibility
+  for (const [oldName, newName] of Object.entries({
```
Comment (fox1t) [inline src/plugins/remote-cache/storage/s3.ts:10]: We can drop this since we will release this as major version bump.

**Review by fox1t — CHANGES_REQUESTED** [review]

File: src/plugins/remote-cache/storage/s3.ts (hunk 1 above)
Comment (dobesv) [reply src/plugins/remote-cache/storage/s3.ts]: OK, I removed that 👍

**Comment by dobesv** [comment]
There seems to be an issue with the build that I'm not sure what to do with, can you give any advice on how to fix that?

**Comment by tehKapa** [comment]
> There seems to be an issue with the build that I'm not sure what to do with, can you give any advice on how to fix that?

are you aligned with the last commit?

**Comment by dobesv** [comment]
> are you aligned with the last commit?

I was ... I'll rebase on main and try again, see if it helps

**Comment by fox1t** [comment]
@dobesv now it is working again. Would you mind modifying tests to match the new implementation?

**Comment by dobesv** [comment]
I updated the env vars for the tests, they pass locally now.  🤞

**Review by fox1t — APPROVED** [review]
LGTM

**Comment by dobesv** [comment]
Any idea when this might land in docker ?

**Pull request merged**